cmd/collector/app/flags/flags.go

```diff
@@ -202,6 +207,8 @@ func addHTTPFlags(flags *flag.FlagSet, cfg serverFlagsConfig, defaultHostPort st
 	flags.Duration(cfg.prefix+"."+flagSuffixHTTPIdleTimeout, 0, "See https://pkg.go.dev/net/http#Server")
 	flags.Duration(cfg.prefix+"."+flagSuffixHTTPReadTimeout, 0, "See https://pkg.go.dev/net/http#Server")
 	flags.Duration(cfg.prefix+"."+flagSuffixHTTPReadHeaderTimeout, 2*time.Second, "See https://pkg.go.dev/net/http#Server")
+	flags.String(cfg.prefix+"."+flagSuffixHTTPAllowedHeaders, "content-type", "Allowed headers for the OTLP HTTP port , default content-type")
+	flags.String(cfg.prefix+"."+flagSuffixHTTPAllowedOrigins, "*", "Allowed origins for the OTLP HTTP port , default accepts all")
```

Comment:
This could add these flags at the top level, e.g. `--collector.allowed-headers`, which is not what we want.

I would rather do a little bit of refactoring and create a new package `pkg/config/corscfg`, similar to `tlscfg`, which will define a shared structure & flags for these parameters. We can then use the new package in both Zipkin endpoint and OTLP endpoint.

We can also reuse OTEL's `confighttp.CORSSettings` as the underlying structure for these params.

Reply:
Flags at the top level as in while calling the flag or internally? because addHTTPFlags is a common function for both collector http port and the otlp server http port and is called with respective server flags. Correct me if I am wrong.

Yes refactoring in a similar way like tlscfg seems like a nice decision will get on the work with that!